src/history/hs.c

```diff
@@ -462,6 +464,14 @@ __hs_insert_record_with_btree_int(WT_SESSION_IMPL *session, WT_CURSOR *cursor, W
             if (cmp == 0)
                 counter = hs_counter + 1;
         }
+
+        /* Now handle the out-of-order case. */
+        if (upd->start_ts != WT_TS_NONE) {
+            WT_ERR_NOTFOUND_OK(cursor->next(cursor), true);
```

Comment:
We're guaranteed to be on the timestamp that we're inserting or JUST before that. If we do a `next`, we'll be on the beginning of the range that we need to fixup.

Reply:
This comment threw me a bit, since I thought in one case we didn't want to move forwards, but what you are saying is that if we land on a record with a matching timestamp, we still want to move forward to the `next` one before starting to fix things up?

Reply:
Add more explanation.